### ISSUE TITLE
Details about map tokens

### ISSUE THREAD

**Issue by OrangeSodahub** [rainmaker22/SMART#16]
Hi, I want to figure out some details about your map tokens.

I checked the `map_traj_token5.pkl` file which contains two keys:

```
"traj_src": shape (1024, 11, 2),
"sample_pt": shape (1024, 3, 2)
```

Could you explain more about what these attributes stand for? I'm confused about the second dimension: `11` and `3`. Here `3` also comes from the map tokenization process.

And what is the difference between `traj_src` and `sample_pt`, could you give some advice?

cc @rainmaker22 @ziyanGao

**Comment by OrangeSodahub** [comment]
Another question is, next token prediction for **map** seems not be used in current version of codes. Is that right?

**Comment by rainmaker22** [comment]
- 'traj_src' represents the original trajectory points of the map tokens, storing the map's original features.
- 'sample_pt' refers to three equidistant points sampled from 'traj_src', used in the 'match_token_map' function to compute the distance to the map’s original trajectory.
Map token pretraining was done separately, and the open-source version omits the loss calculation. You can add a cross-entropy loss for map tokens in the 'forward' function, similar to NTP's cls loss, for pretraining.

**Comment by OrangeSodahub** [comment]
@rainmaker22 Thanks for your useful information! So the 'traj_src' and 'sample_pt' both represent the 0.5m-long map trajectories, and 'traj_src' has 11 points while 'sample_pt' has 3 points.

- But where is `11` from? Can I consider the tokenization of map is the same as the agent trajectory? while you use 11 points in map v.s. use 5 steps in agent
- Can I say that using `3` is a trade off between precision and computation?
- Have you ever try joint training? I wonder how can pretraining bring improvement.

**Comment by OrangeSodahub** [comment]
Another question is, if you pretrained map decoder. When training agent decoder, did you freeze the pretrained map decoder or not? Following the current version of codes, it seems that you didn't freeze.

**Comment by rainmaker22** [comment]
The choice of 11 points is a design decision that provides a reasonable level of detail: each map token covers around 5m, so 11 points (approximately 0.5m spacing) effectively capture map features. In contrast, agent tokens are fixed in time (0.5s intervals), so we use points sampled at 0.1s intervals to represent trajectories. The choice of 3 points is indeed a trade-off between precision and computational efficiency.
Regarding pretraining, map decoder pretraining is mainly beneficial for cross-dataset zero-shot tasks, and can generally be disregarded when using single datasets. Joint training yields similar results to pretraining alone. During agent training, the map encoder is not frozen.
